tests/unit/dialog/dialog_core.js

```diff
@@ -7,6 +7,37 @@
 // TODO add teardown callback to remove dialogs
 module("dialog: core");
 
+test( "markup structure", function() {
+	expect( 7 );
+	var element = $("<div></div>").dialog({
+			buttons: [{
+				text: "Ok",
+				click: $.noop
+			}]
+		}),
+		widget = element.dialog( "widget" ),
+		titlebar = widget.find( ".ui-dialog-titlebar" ),
+		close = titlebar.find( "button" ),
+		buttonpane = widget.find( ".ui-dialog-buttonpane" );
+
+	ok( widget.is( ".ui-dialog.ui-widget.ui-widget-content.ui-corner-all.ui-dialog-buttons" ),
+		"Widget has proper classes" );
+
+	ok( titlebar.is( ".ui-dialog-titlebar.ui-corner-all.ui-widget-header"),
+		"Titlebar has proper classes");
+
+	ok( titlebar.find( ".ui-dialog-title" ).length, "Title is .ui-dialog-title" );
+
+	ok( close.hasClass( "ui-dialog-titlebar-close" ), "Close button is .ui-dialog-titlebar-close" );
+
+	ok( element.is( ".ui-dialog-content.ui-widget-content" ), "Element is .ui-widget-content" );
+
+	ok( buttonpane.is( ".ui-dialog-buttonpane.ui-widget-content" ),
+		"Buttonpane is .ui-widget-content" );
+	ok( buttonpane.find( ".ui-dialog-buttonset" ).length, "buttonset is .ui-dialog-buttonset" );
+
+});
```

Comment:
So I know we've gone back and forth on this. I'd be ok with changing these tests to check against the full `class` attribute so we can see the diff on a failure, but I'd also say these can just be many `.hasClass()` tests for each individual class. Right now what bothers me is the inconsistency in messages.